sql/src/test/java/io/crate/planner/operators/PushDownTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.analyze.TableDefinitions;
+import io.crate.planner.TableStats;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Test;
+
+import static io.crate.planner.operators.LogicalPlannerTest.isPlan;
+
+public class PushDownTest extends CrateDummyClusterServiceUnitTest {
+
+    @Test
+    public void pushDownUnionOrder() {
+
+        TableStats tableStats = new TableStats();
+
+        SQLExecutor sqlExecutor = SQLExecutor.builder(clusterService)
+            .addDocTable(TableDefinitions.USER_TABLE_INFO)
+            .build();
+
+        LogicalPlan plan = LogicalPlannerTest.plan(
+            "Select name from users union all select text from users order by name",
+            sqlExecutor,
+            clusterService,
+            tableStats);
+
+        assertThat(plan, isPlan(sqlExecutor.functions(),  "Boundary[name]\n" +
+                                                "Union[\n" +
+                                                    "Boundary[name]\n" +
+                                                    "OrderBy[name]\n" +
+                                                    "Collect[doc.users | [name] | All]\n" +
+                                                "---\n" +
+                                                    "Boundary[text]\n" +
+                                                    "OrderBy[text]\n" +
+                                                    "Collect[doc.users | [text] | All]\n" +
+                                                "]\n"));
+    }
+
+    @Test
+    public void pushDownUnionWithTwoOrders() {
```

Comment:
I'd rename it to `testOrderByOnUnionIsMovedAndOverwritesChildRelationOrderBy` or something like that.

Reply:
Maybe a comment is a better here.